packages/synapse-react-client/src/utils/functions/TableColumnSchemaUtils.ts

```diff
@@ -47,23 +53,33 @@ export async function createTableUpdateTransactionRequest(
     .list
 
   // now that all columns have been created, figure out the column changes (create, update, and no-op)
-  // keep track of column ids to figure out what columns were deleted.
   const changes: ColumnChange[] = []
-  const columnIds: Set<string | null> = new Set()
+  const columnIdsThatWereChangedOrAdded: Set<string> = new Set()
   for (let i = 0; i < proposedSchema.length; i++) {
     const oldColumnId: string | null = proposedSchema[i].id ?? null
     const newColumnId: string | null = createdColumnModels[i].id ?? null
-    columnIds.add(oldColumnId)
-    columnIds.add(newColumnId)
-    if (oldColumnId !== newColumnId) {
+    if (oldColumnId) columnIdsThatWereChangedOrAdded.add(oldColumnId)
+    columnIdsThatWereChangedOrAdded.add(newColumnId)
+    if (
+      // The column has changed
+      oldColumnId != null &&
+      oldColumnId !== newColumnId
+    ) {
       changes.push({ oldColumnId, newColumnId })
+    } else if (
+      // the column is new and is defined by the user
+      oldColumnId == null ||
+      // the column is new but already has an ID (e.g. we are adding a default column to a view)
+      !oldColumnModelId2Model.has(oldColumnId)
```

Comment:
We were missing this case before. The editor would fail to add a change for default view columns. The check `oldColumnId !== newColumnId` assumed that all of the `oldColumnId` values would be in the `oldSchema`, but that was not true for default view columns.